File: 2379-maximum-total-importance-of-roads/2379-maximum-total-importance-of-roads.java
class Solution {
    public long maximumImportance(int n, int[][] roads) {
        int degree[] = new int[n];
        for (int a[]: roads){
            degree[a[0]]++;
            degree[a[1]]++;
        }

        Arrays.sort(degree);
        long val =1;
        long ans =0;
        for(int i =0;i<n;i++){
            ans += val * degree[i];
            val++;
        }

        return ans;
    }
}

File: 2379-maximum-total-importance-of-roads/2379-maximum-total-importance-of-roads.go
func maximumImportance(n int, roads [][]int) int64 {
   degree := make([]int, n)
   for i:=0 ;i<len(roads) ; i++{
     degree[ roads[i][0]]++
     degree[roads[i][1]]++
   }
   sort.Ints(degree)
   var val int64 =1
   var ans int64 = 0
   for i:=0;i<len(degree);i++{
     k := val  *  int64(degree[i])
     ans = ans+ k
     val= val+1
   }
    return ans
}